src/experimental-layers/src/solid-polygon-layer/polygon.js

```diff
@@ -80,9 +84,38 @@ export function forEachVertex(polygon, visitor) {
     return;
   }
 
-  let vertexIndex = 0;
   polygon.forEach(simplePolygon => {
-    simplePolygon.forEach((v, i, p) => visitor(v, vertexIndex, polygon));
-    vertexIndex++;
+    simplePolygon.forEach(visitor);
   });
 }
+
+// Returns the offset of each hole polygon in the flattened array for that polygon
+function getHoleIndices(complexPolygon) {
+  let holeIndices = null;
+  if (complexPolygon.length > 1) {
+    let polygonStartIndex = 0;
+    holeIndices = [];
+    complexPolygon.forEach(polygon => {
+      polygonStartIndex += polygon.length;
+      holeIndices.push(polygonStartIndex);
+    });
+    // Last element points to end of the flat array, remove it
+    holeIndices.pop();
+  }
+  return holeIndices;
+}
+
+/*
+ * Get vertex indices for drawing complexPolygon mesh
+ * @private
+ * @param {[Number,Number,Number][][]} complexPolygon
+ * @returns {[Number]} indices
+ */
+export function getSurfaceIndices(complexPolygon) {
```

Comment:
This moves the core tesselation from the "tesselator" into a basic polygon utility file. Changes the "flavor" of this file but that's OK, doesn't affect anything else..